c-dependencies/js-compute-runtime/rust-url/src/lib.rs

```diff
@@ -36,13 +36,26 @@ impl JSUrlSearchParams {
         match self.url_or_str {
             UrlOrString::Url(url) => {
                 let url = unsafe { url.as_mut().unwrap() };
-                let mut pairs = url.url.query_pairs_mut();
-                pairs.clear().extend_pairs(self.list.iter());
+                match self.list.is_empty() {
+                    true => {
+                        url.url.set_query(None);
```

Comment:
This is the crux of the bug-fix.
Previously we used `url.query_pairs_mut().clear()` however `url.query_pairs_mut().clear();` is equivalent to `url.set_query(Some(""))`, not `url.set_query(None)` and we really wanted `.set_query(None)`